src/components/calculator/SavedDataSelector.tsx

```diff
@@ -48,19 +65,47 @@ export const SavedDataSelector: React.FunctionComponent<{
         <Typeahead
           clearButton={true}
           emptyLabel={t('calculator.no_prebuilt_scenario_found')}
-          multiple
+          filterBy={showMatchesAndCustomScenario}
           highlightOnlyResult={true}
           id="predefined-typeahead"
-          inputProps={{ autoComplete: 'chrome-off' }}
+          inputProps={{
+            autoComplete: 'chrome-off',
+            shouldSelectHint: (shouldSelect: boolean, event) => {
+              if (event.keyCode === 13 /* return */) {
+                return true
+              }
+              return shouldSelect
+            },
+          }}
           onChange={(e: string[]) => {
             if (e.length === 0) {
               setSavedData('')
             }
             setSavedData(e[e.length - 1])
           }}
+          onBlur={() => {
+            if (selected2.length === 0) {
+              selected2 = [i18n.t('scenario.custom')]
+              if (savedDataRef.current) {
+                savedDataRef.current.getInput().value = i18n.t(
+                  'scenario.custom',
+                )
+              }
+              setSavedData(i18n.t('scenario.custom'))
```

Comment:
Is this saying that, if I click into the typeahead and then change my mind, it will wipe out any data I've entered?

Reply:
if you tab through in any configuration (preset, modified from preset, custom blank, custom blank + some modifications), there is no change.

If you click/tab in, then click out, there is no change.

If you click/tab in, hit a few arrow keys, tab brings you out with no change. (same behaviour as current site; needs typescript definition update)

If you click/tab in, and hit any keyboard key other than tab/enter/arrows, it does clear the form on the way out. I suppose it's possible to add a check to see if the input bar text is empty, and if it's empty, don't reset.

Reply:
what this is *for* is 

I type in "hair cut" and it tells me I can build my own scenario. Enter already works, but tab did not: tab used to silently focus the next input element on the page. That means that if you had previously selected "one night stand" and then started typing hair cut, you would see a prompt to build your own scenario. If you then hit tab, you'd be in a weird spot where

* the precautions section is gone
* there's a risk level still showing for your "hair cut"
* you saw a suggestion to build your own and acted on it and have every reason to believe you are building your own hair cut scenario

Reply:
Here's a summary of the current and new behaviour in a table. Origin/Notes section is filled out for every change in behaviour (including internal behaviour that results in the same outward behaviour).

Do you still have concerns?
Would you be ok with deferring G for a future PR?

Setup | Bar autocompletes? | Current behaviour | New behaviour | Origin/Notes |
------ | ------------- | ---------------- | ---------------- | --- |
A. Type "outdoor", hit tab | Yes | Selects first entry | Selects first entry
B. Type "outdoor", hit enter | Yes | Nothing happens :x: | Selects first entry | shouldSelectHint
C. Type "masked", hit tab | No | Focus next element | Select custom scenario, reset form, focus next element | OnBlur
D. Type "masked", hit enter | No | Nothing | Nothing
E. Type "water", hit tab | No | Focus next element | Select custom scenario, reset form, focus next element | OnBlur
F. Type "water", hit enter | No | Nothing | Select custom scenario, reset form, focus next element | highlightOnlyResult=true
G. Focus bar, optionally type "masked", use arrow keys, hit tab | Yes | Keeps last selection :x: | Works. Uses new selection that's already highlighted and in the bar autocomplete hint. | Pending upstream change; not uploaded yet
H. Focus bar, optionally type "masked", use arrow keys, hit enter | Yes | Works | Works
I. No existing selection, focus in, optional highlight with mouse, focus out | | No selection change | No selection change | onBlur checks selected.length before acting
J. Existing selection, focus in, optional highlight with mouse, focus out | | No selection change | No selection change | onBlur checks selected.length before acting